src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx

```diff
@@ -407,8 +413,33 @@ export const useTradeExecution = (
           cancelPollingRef.current = output?.cancelPolling
           return
         }
-        case CHAIN_NAMESPACE.Solana:
+        case CHAIN_NAMESPACE.Solana: {
+          const adapter = assertGetSolanaChainAdapter(stepSellAssetChainId)
+          const from = await adapter.getAddress({ accountNumber, wallet })
+          const output = await execution.execSolanaTransaction({
+            swapperName,
+            tradeQuote,
+            stepIndex: hopIndex,
+            slippageTolerancePercentageDecimal,
+            from,
+            signAndBroadcastTransaction: async (txToSign: SolanaSignTx) => {
+              const signedTx = await adapter.signTransaction({
+                txToSign,
+                wallet,
+              })
+              const output = await adapter.broadcastTransaction({
+                senderAddress: from,
+                receiverAddress: tradeQuote.receiveAddress,
+                hex: signedTx,
+              })
+
+              trackMixpanelEventOnExecute()
+              return output
+            },
+          })
+          cancelPollingRef.current = output?.cancelPolling
```

Comment:
Thanks for this, I'll need all those Solana bits for Jup, I started but I'll rebase on this branch to retrieve your work 💜